### PR TITLE
IoTCentral: Add and swap to embedded SDK

### PR DESCRIPTION
**Pull request by stephybun**

### CODE REVIEW

File: internal/services/iotcentral/iotcentral_application_resource.go [hunk 1]
```diff
@@ -73,12 +72,11 @@ func resourceIotCentralApplication() *pluginsdk.Resource {
 				Type:     pluginsdk.TypeString,
 				Optional: true,
 				ValidateFunc: validation.StringInSlice([]string{
-					string(iotcentral.F1),
-					string(iotcentral.S1),
-					string(iotcentral.ST1),
-					string(iotcentral.ST2),
+					string(apps.AppSkuSTOne),
+					string(apps.AppSkuSTTwo),
+					string(apps.AppSkuSTZero),
```
Comment (tombuildsstuff) [inline internal/services/iotcentral/iotcentral_application_resource.go:77]: F1 isn't present here?

File: internal/services/iotcentral/iotcentral_application_resource.go [hunk 2]
```diff
@@ -73,12 +72,11 @@ func resourceIotCentralApplication() *pluginsdk.Resource {
...
 				}, false),
-				Default: iotcentral.ST1,
+				Default: apps.AppSkuSTOne,
```
Comment (tombuildsstuff) [inline internal/services/iotcentral/iotcentral_application_resource.go:79]: fwiw these should be:

```suggestion
				Default: string(apps.AppSkuSTOne),
```

File: internal/services/iotcentral/iotcentral_application_resource.go [hunk 3]
```diff
@@ -98,61 +96,56 @@ func resourceIotCentralAppCreate(d *pluginsdk.ResourceData, meta interface{}) er
...
 	}
 
-	resp, err := client.CheckNameAvailability(ctx, iotcentral.OperationInputs{
-		Name: utils.String(name),
-	})
+	inputs := apps.OperationInputs{
+		Name: id.ResourceName,
+	}
+
+	resp, err := client.CheckNameAvailability(ctx, commonids.SubscriptionId{SubscriptionId: id.SubscriptionId}, inputs)
```
Comment (tombuildsstuff) [inline internal/services/iotcentral/iotcentral_application_resource.go:115]:
```suggestion
	resp, err := client.CheckNameAvailability(ctx, commonids.NewSubscriptionID(id.SubscriptionId), inputs)
```

File: internal/services/iotcentral/iotcentral_application_resource.go [hunk 4]
```diff
@@ -199,55 +188,54 @@ func resourceIotCentralAppRead(d *pluginsdk.ResourceData, meta interface{}) erro
...
+	return nil
 }
 
 func resourceIotCentralAppDelete(d *pluginsdk.ResourceData, meta interface{}) error {
 	client := meta.(*clients.Client).IoTCentral.AppsClient
 	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := parse.ApplicationID(d.Id())
+	id, err := apps.ParseIotAppIDInsensitively(d.Id())
```
Comment (tombuildsstuff) [inline internal/services/iotcentral/iotcentral_application_resource.go:232]:
```suggestion
	id, err := apps.ParseIotAppID(d.Id())
```

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
Left a few comments inline - will follow up offline 👍

File: internal/services/iotcentral/iotcentral_application_resource.go (hunk 1 above)
Comment (stephybun) [reply internal/services/iotcentral/iotcentral_application_resource.go:77]: The new version of the SDK doesn't have F1 or S1 anymore. Even though the older SDK has F1 and S1 as valid enums the API will throw an error if you try and create this resource with F1 or S1.

**Comment by hmmorales** [comment]
Hello is there an expected date for this PR? We are curious because [16899](https://github.com/hashicorp/terraform-provider-azurerm/pull/16899) is blocked until this PR is merged.

**Pull request merged**